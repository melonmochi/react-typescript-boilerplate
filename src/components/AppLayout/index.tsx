import * as React from "react";
import { useTranslation } from "react-i18next";
import ProLayout from "@ant-design/pro-layout";

const AppLayout: React.FC = () => {
  const [t, i18n] = useTranslation();
  return <ProLayout title={t("global.title")} />;
};

export default AppLayout;
