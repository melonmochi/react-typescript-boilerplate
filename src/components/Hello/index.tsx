import * as React from "react";
import { Button } from "antd";

export interface HelloProps {
  compiler: string;
  framework: string;
}

const Hello = (props: HelloProps) => (
  <h1>
    <Button type="primary">Button</Button>
    Hello from {props.compiler} and {props.framework}!
  </h1>
);

export default Hello;
