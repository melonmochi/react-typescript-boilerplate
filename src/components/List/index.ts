export { default as TableList } from "./TableList";
export { BigDataTableList } from "./BigDataTableList";
