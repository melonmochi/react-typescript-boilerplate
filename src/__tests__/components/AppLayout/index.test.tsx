import * as React from "react";
import { shallow } from "enzyme";
import { AppLayout } from "components";

describe("AppLayout", () => {
  test("Snapshot", () => {
    const appLayout = shallow(<AppLayout />);
    expect(appLayout).toMatchSnapshot();
  });
});
