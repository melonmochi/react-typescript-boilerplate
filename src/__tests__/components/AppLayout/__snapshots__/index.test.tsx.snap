// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoFoundPage Snapshot 1`] = `
<BasicLayout
  autoHideHeader={false}
  contentWidth="Fluid"
  fixSiderbar={false}
  fixedHeader={false}
  iconfontUrl=""
  layout="sidemenu"
  location={
    Location {
      "assign": [Function],
      "hash": "",
      "host": "localhost",
      "hostname": "localhost",
      "href": "http://localhost/",
      "origin": "http://localhost",
      "pathname": "/",
      "port": "",
      "protocol": "http:",
      "reload": [Function],
      "replace": [Function],
      "search": "",
      "toString": [Function],
    }
  }
  logo="https://gw.alipayobjects.com/zos/antfincdn/PmY%24TNNDBI/logo.svg"
  menu={
    Object {
      "locale": true,
    }
  }
  navTheme="dark"
  title="global.title"
/>
`;
